### ISSUE TITLE
if operator forgot to start the timer

### ISSUE THREAD

**Issue by hoczykm** [AntonyCorbett/OnlyT#73]
Hello. Common issue in our congregation is that the operator forgets to start the timer due to many other tasks he's responsible for. I think that if the application somehow catches eye when the timer is not running, it will significantly reduce such incidents. My proposal is to apply to OnlyT background some color (e.g. light red? ) if the timer is not running, and standard color - if the timer is running.

**Comment by donavan1** [comment]
Hi, Great idea, or even blink or a text.

**Comment by Blann** [comment]
Problem with this is that there’s no way for OnlyT to know when the timer should be running and when it shouldn’t. It worked in SoundBox because (I’m guessing) it could extrapolate based on when the songs were played. Something flashing constantly because the timer isn’t running could get annoying/distracting if you just want OnlyT open to display the time of day, for example.

**Comment by donavan1** [comment]
Just a suggestion, can this only be on the main pc on the app as it
currently as a timer. that is being controlled by the sound department, it
should not be displayed on the timer screen.

**Comment by donavan1** [comment]
thats perfect

**Comment by josslamberti** [comment]
While I'm trying to think about how this would be achievable, one reasonable workaround is to delegate the timing task to another person through the mobile app. After all, the main issue here is that the sound assistant has many jobs to do and delegating is the best way to fix multitasking issues.

**Comment by AntonyCorbett** [comment]
@josslamberti thanks, that's a useful observation.

**Comment by Chipperchap** [comment]
I hope I am not introducing another topic within this one, but rather expanding it.  

Our brothers have become accustomed to having the timer to let them know how their talk duration is going.  However on occasion our booth operators have forgotten to start the timer but then remembers maybe 2 or 3 minutes into, for example, the TFGW part. If the booth operator then clicks on 'start', then it counts down from 10 minutes even though there is now actually only 8 or 7 minutes left.  

Could there therefore be a way to choose to decrease the time allotted? I see that within 'Manual' mode one can increase/decrease the timer (for obvious reasons) but this is always greyed out for Automatic mode.  Possibly the increase/decrease toggles could be optional?  

This might also eliminate the need for blinking buttons etc. as mentioned above?

**Comment by AntonyCorbett** [comment]
>  always greyed out for Automatic mode

Actually some of the timers _can_ be modified in Automatic mode. Other items always have a fixed duration in the schedule so the adjustment buttons are grayed out. But I understand your point - I'll have a think...

**Comment by ucphinni** [comment]
Would it possible to have an adjust button for absolute time?  This should only be active when the timer is currently active.

In general, the sound attendant does know what Wall Clock time the last timer should have started, not how much time needs to be taken off.

Work flow:
1. SA starts timer late.
2. Timer adjust (new) button gets activated.
3. SA clicks on this button.  Dialog comes up presenting clock (hours, minutes seconds)
4. SA sets the start time and hits ok on dialog.
5. Timer jumps to appropriate time.  This even works with adaptive timers.

**Comment by AntonyCorbett** [comment]
Thanks for ideas on "forgottem timer" issue - keep them coming

**Comment by josslamberti** [comment]
Is there a technical block to adjusting the remaining time while the timer is running? 

This is my own observation and maybe I'm wrong but I'd say that when the sound attendant realises that the timer is not running when it should, he will probably, in a panic, start the timer first THEN think about how much time may have already elapsed since the item has started. He may then remove (or add but that's less likely to happen) chunks of 30s (1 min may be too much) on the fly.
Once the timer has started, if you can't adjust it, then you need to stop it, reselect the item, remove some more time, restart it. 

This is a similar request to the one above (adjusting while running), however regarding @ucphinni 's suggestion, I wouldn't rely on knowledge of the current time when the sound attendant already has to juggle with so much. 

As for estimating how much time may have actually elapsed, I'd suggest a secondary, automatic, timer that would start every time the main countdown has stopped and would disappear as soon as the main timer is started. It would be a good help to estimate the elapsed time since the beginning of the current item and even show an extra visual clue that the timer is not running.

Example:
At the end of the 10 min Treasures talk, I forgot to start the timer. The running count-up timer below would tell me that I have probably forgotten to start the timer because more than 2 minutes have already elapsed.

![image](https://user-images.githubusercontent.com/30600877/48832891-1fc72680-ed72-11e8-89cb-871bb171eadd.png)

Some observations:
- the 'yellow' timer doesn't need to be shown straight away, it may be shown after 30 seconds only (value to be discussed) but it still needs to start in the background as soon as possible
- if adjusting the timer while it's running is possible, the 'yellow' timer should NOT be hidden as soon as the main timer has started, because the sound assistant needs to know how much time he may remove 
- the 'yellow' timer should only be shown to the sound assistant. Showing it to the speaker would be confusing

Now if I start the main timer after 2min14s, as shown in the example, then perhaps we could have an option to remove automatically 1min44s from the current timer. 

![image](https://user-images.githubusercontent.com/30600877/48833623-ad574600-ed73-11e8-91b7-1091e733ad69.png)




Whether the above is possible or not, for a start, I would suggest allowing all items' times to be manually adjustable in automatic mode. I believe this should be a simple change to the automatic mode feed, using the editable flag on all items.

**Comment by Blann** [comment]
@josslamberti There’s probably not a technical block, but it would be a bit disconcerting/confusing for the brother on the platform to see his remaining time changing by 30 seconds at a time if he just glanced at it. 

My own view is that, while some of these ideas are technically possible/workable, they massively overcomplicate the interface and core functionality. In my experience anything beyond a simple start/stop button is confusing for a not-inconsiderable number of operators. The simplest solution would just be a prompt (whatever form that takes) to start the timer at the correct time.

**Comment by josslamberti** [comment]
Extra note about my suggestion above: 
If adjusting the timer while it's running is not possible, then maybe we could have a two-in-one action to remove the suspected missed 1m44 and then start the timer.

**Comment by ucphinni** [comment]
Agreed but mistakes do happen and can be caught semi-gracefully if onlyt can support it. Quite a few scenarios cause missed timer starts from my experience and observations:
• Remote timer operating doing multiple duties or not there that day (e.g. platform) and misses cues
• Network unavailability for remote timer support 
• System or onlyt Crash
• SA operator or remote operator falls asleep (yes this happened)

To help mitigate your complexity concerns
• Last timer start can be added to the settings panel. On opening of the settings panel, adjustable start time would be set to the last timer start time, regardless of whether the timer is running on not.  When settings are closed, the start time would be applied to the running timer.  This encourages only advanced special case usage.
• Although I am not a big fan of this, an indication that an adjustment is made to the platform temporarily thru the web interface and the standard display via animation, icon, or flashing.

**Comment by garethawhite** [comment]
Could there just be an indication (flashing indicator?) when the timer is NOT timing an item. i.e. When it is PENDING, waiting to start timing the next item. This might be enough keep the operator aware that OnlyT is waiting for him to click the start button.

**Comment by garethawhite** [comment]
Additional thought to my suggestion above: Perhaps what would remind the operator to click START could be a blinking arrow (or other symbol) along side the Start Button, or a blinking frame around the Start Button.

**Comment by josslamberti** [comment]
@AntonyCorbett would it be possible to make all items editable in the Automatic mode? That'd be a good (and easy) start for working around this issue.

**Comment by eatyourpeas747** [comment]
I think it could be helpful for on the remote app/main program for there to be a button that says "Late Start" or something and that starts the countdown but the clock color is orange or so that the speaker will know that they need to cut the time down even more.

**Comment by AntonyCorbett** [comment]
@eatyourpeas747 thanks for your suggestion. Doesn't the adaptive mode provide what's needed here?

**Comment by eatyourpeas747** [comment]
> @eatyourpeas747 thanks for your suggestion. Doesn't the adaptive mode provide what's needed here?
I don't see a way for someone using only the app to start the timer to adjust the time.  Unless Im missing something

**Comment by AntonyCorbett** [comment]
@josslamberti and others - Thanks for all your ideas. I have implemented a "reminder pop-up" feature in the latest pre-release (v2.3.0.1 onwards). This works by displaying a simple reminder notification bottom right of the primary monitor when OnlyT thinks you may have forgotten to start a timer. I'm sure the feature could be more sophisticated and there may be some tweaking to do, but I hope it's of some use. It's on by default.
